(dp1
S'output'
p2
S''
sS'layer'
p3
S'/var/www/web2py/applications/favoritesmanager/controllers/bookmark.py'
p4
sS'code'
p5
S"# -*- coding:utf-8 -*-\n'''\nController para bookmarks onde terao todas as funcoes para trabalhar\ncom os marcadores\n'''\ndef index():\n\tredirect(URL('new'))\n\n@auth.requires_login()\t\ndef new():\n\tform = SQLFORM(db.bookmarks, fields=['name','url'])\n\t\n\tif form.accepts(request.vars, session):\n\t\tdb(db.bookmarks).update(user = auth.user.email, creation = request.now)\n\t\tresponse.flash = T('New bookmark was created !')\n\telif form.errors:\n\t\tresponse.flash = T('Oops, look whos wrong !')\n\n\treturn {'form':form}\n\t\n@auth.requires_login()\ndef show():\n\tmymarks = db(db.bookmarks).select()\n\treturn {'marks':mymarks}\n\n@auth.requires_login()\ndef remove():\n\tif request.args(0):\n\t\tdb(db.bookmarks.id == request.args(0)).delete()\n\t\tredirect(URL('bookmark','show'));\n\telse:\n\t\tredirect(URL('bookmark','show'))\n\ndef edit():\n\trec = db.person(request.args(0)) or redirect(URL('show'))\n\tform = SQLFORM(db.bookmarks, rec,fields=['name','url'])\n\t\n\tif form.accepts(request.vars, session):\n\t\tresponse.flash = T('Bookmark up-to-date !')\n\telif form.errors:\n\t\tresponse.flash = T('Oops, look whos wrong !')\n\n\treturn {'form':form}\n\nresponse._vars=response._caller(edit)\n"
p6
sS'snapshot'
p7
(dp8
sS'traceback'
p9
S'Traceback (most recent call last):\n  File "/var/www/web2py/gluon/restricted.py", line 192, in restricted\n    exec ccode in environment\n  File "/var/www/web2py/applications/favoritesmanager/controllers/bookmark.py", line 45, in <module>\n  File "/var/www/web2py/gluon/globals.py", line 145, in <lambda>\n    self._caller = lambda f: f()\n  File "/var/www/web2py/applications/favoritesmanager/controllers/bookmark.py", line 35, in edit\n    rec = db.person(request.args(0)) or redirect(URL(\'show\'))\n  File "/var/www/web2py/gluon/dal.py", line 4331, in __getattr__\n    return self[key]\n  File "/var/www/web2py/gluon/dal.py", line 4325, in __getitem__\n    return dict.__getitem__(self, str(key))\nKeyError: \'person\'\n'
p10
s.